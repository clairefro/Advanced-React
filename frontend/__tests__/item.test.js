import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: 'ABC123',
  title: 'A Cool Item',
  price: 5000,
  description: 'so cool item',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg'
}

describe('<Item />', () =>{
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  // it('renders the image properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const image = wrapper.find('img');
  //   console.log(image.props());
  //   expect(image.props().src).toBe(fakeItem.image);
  //   expect(image.props().alt).toBe(fakeItem.title);
  // });
  //
  // it('renders the price tag and title properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   console.log(wrapper.debug());
  //   const PriceTag = wrapper.find('ForwardRef(PriceTag)');
  //   expect(PriceTag.text()).toBe('$50');
  //   // console.log(wrapper.find('ForwardRef(Title) a').debug());
  //   expect(wrapper.find('ForwardRef(Title) a').text()).toBe(fakeItem.title);
  // });
  //
  // it('renders buttons', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   console.log(wrapper.debug());
  //   const buttonList = wrapper.find('.buttonList')
  //   console.log(buttonList.children());
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link').exists()).toBe(true);
  //   expect(buttonList.find('AddToCart').exists()).toBe(true);
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true);
  // });
});
