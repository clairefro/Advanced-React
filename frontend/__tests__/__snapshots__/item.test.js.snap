// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ForwardRef(ItemStyles__Item)>
  <img
    alt="A Cool Item"
    src="dog.jpg"
  />
  <ForwardRef(Title)>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </ForwardRef(Title)>
  <ForwardRef(PriceTag)>
    $50
  </ForwardRef(PriceTag)>
  <p>
    so cool item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete
    </DeleteItem>
  </div>
</ForwardRef(ItemStyles__Item)>
`;
